### PR TITLE
Return early when channels mismatch to skip expensive msgpack decoding

### PR DESCRIPTION
**Pull request by ryoqun**
Hopefully, the title says all.

Well, we're using a lot of namespaces. and there is performance problem.

### CODE REVIEW

**Comment by darrachequesne** [comment]
LGTM :+1: 

Could you squash it into one commit please?

**Comment by ryoqun** [comment]
@darrachequesne squashed! Thanks for quick reply!

(hopefully, version bump is preferred... :p)

**Comment by ryoqun** [comment]
@nkzawa @rauchg (Kindly pinging to just remind this project's members; sorry for being impatient! ;) )

File: index.js [hunk 1]
```diff
@@ -68,11 +68,21 @@ function adapter(uri, opts){
...
     this.prefix = prefix;
+    this.channel = prefix + '#' + nsp.name + '#';
+    if (String.prototype.startsWith) {
+      this.channelMatches = function (messageChannel, subscribedChannel) {
+        return messageChannel.startsWith(subscribedChannel);
+      }
+    } else { // Fallback to slow indexOf impl for older Node.js
+      this.channelMatches = function (messageChannel, subscribedChannel) {
+        return messageChannel.indexOf(subscribedChannel) === 0;
+      }
```
Comment (rauchg) [inline index.js]: How about just using `substr` ?

Comment (ryoqun) [inline index.js]: @rauchg Fixed! f2e7a4c8d131ce6b704a8daa4d8e10c7d514fc70

Comment (ryoqun) [inline index.js]: @rauchg I referred to https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

I leant something from your suggestion, thank you!

**Comment by rauchg** [comment]
Thanks!

**Comment by ryoqun** [comment]
@rauchg Thanks a lot! Also, could you tell me when a new version will be released? :)

**Comment by ryoqun** [comment]
@darrachequesne @nkzawa @rauchg sorry for indiscriminate pinging... Some time has passed. Hopefully, I rather want this modification to be officially released.

**Pull request merged**